src/test/java/teammates/ui/webapi/GetFeedbackSessionsActionTest.java

```diff
@@ -201,7 +203,213 @@ protected void testExecute_asStudent_shouldReturnAllSessionsForAccount() {
+        StudentAttributes student4InCourse1 = typicalBundle.students.get("student4InCourse1");
+        String emailAddress = student4InCourse1.getEmail();
+        FeedbackSessionAttributes session2InCourse1 = typicalBundle.feedbackSessions.get("session2InCourse1");
+        loginAsStudent(student4InCourse1.getGoogleId());
+
+        Instant newEndTime = Instant.now().plus(Duration.ofHours(-1));
+        logic.updateFeedbackSession(FeedbackSessionAttributes.updateOptionsBuilder(
+                session2InCourse1.getFeedbackSessionName(), session2InCourse1.getCourseId())
+                .withEndTime(newEndTime)
+                .build());
+        List<FeedbackSessionAttributes> expectedSessions = sessionsInCourse1.stream()
+                .map(session -> session.sanitizeForStudent(emailAddress))
+                .collect(Collectors.toList());
+        FeedbackSessionAttributes expectedSession2InCourse1 = expectedSessions.get(
+                expectedSessions.indexOf(session2InCourse1));
+        expectedSession2InCourse1.setEndTime(newEndTime);
+
+        String[] submissionParam = {
+                Const.ParamsNames.ENTITY_TYPE, Const.EntityType.STUDENT,
+        };
+
+        ______TS("Before deadline; should indicate open.");
+
+        GetFeedbackSessionsAction a = getAction(submissionParam);
+        FeedbackSessionsData fsData = (FeedbackSessionsData) getJsonResult(a).getOutput();
+
+        assertAllStudentSessionsMatch(fsData, expectedSessions.subList(0, 4), emailAddress);
+
+        ______TS("After deadline but within grace period; should indicate in grace period.");
+
+        Map<String, Instant> studentDeadlines = expectedSession2InCourse1.getStudentDeadlines();
+        studentDeadlines.put(emailAddress, Instant.now().plusSeconds(-1));
+        logic.updateFeedbackSession(FeedbackSessionAttributes.updateOptionsBuilder(
+                expectedSession2InCourse1.getFeedbackSessionName(), expectedSession2InCourse1.getCourseId())
+                .withStudentDeadlines(studentDeadlines)
+                .build());
+
+        a = getAction(submissionParam);
+        fsData = (FeedbackSessionsData) getJsonResult(a).getOutput();
+
+        assertAllStudentSessionsMatch(fsData, expectedSessions.subList(0, 4), emailAddress);
+
+        ______TS("After deadline and beyond grace period; should indicate closed.");
+
+        studentDeadlines.put(emailAddress, Instant.now().plus(Duration.ofHours(-1)));
+        logic.updateFeedbackSession(FeedbackSessionAttributes.updateOptionsBuilder(
+                expectedSession2InCourse1.getFeedbackSessionName(), expectedSession2InCourse1.getCourseId())
+                .withStudentDeadlines(studentDeadlines)
+                .build());
+
+        a = getAction(submissionParam);
+        fsData = (FeedbackSessionsData) getJsonResult(a).getOutput();
+
+        assertAllStudentSessionsMatch(fsData, expectedSessions.subList(0, 4), emailAddress);
```

Comment:
Sorry can help me understand how "closed" is indicated?

Reply:
If I am not wrong, `assertAllStudentSessionsMatch` already checks if the submission statuses matches up through `assertPartialInformationMatch`.
![image](https://user-images.githubusercontent.com/65202977/163661280-25c2d1b2-b3c9-4a86-8b56-5c654ca82e6c.png)

Reply:
Thanks!